In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import axes3d
import seaborn as sns

from sklearn.preprocessing import scale
import sklearn.linear_model as skl_lm
from sklearn.metrics import mean_squared_error, r2_score
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [7]:
advertising = pd.read_csv('./Advertising.csv', usecols=[1,2,3,4])
advertising.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   TV         200 non-null    float64
 1   Radio      200 non-null    float64
 2   Newspaper  200 non-null    float64
 3   Sales      200 non-null    float64
dtypes: float64(4)
memory usage: 6.4 KB


## Regression between sales and radio

In [11]:
# Defining the independent variable (Radio) and the dependent variable (Sales)
X = advertising['Radio']
y = advertising['Sales']

# Adding a constant to the independent variable (for the intercept)
X = sm.add_constant(X)

# Creating the linear regression model
model = sm.OLS(y, X).fit()

# Getting the model summary
model_summary = model.summary()
model_summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.332
Model:                            OLS   Adj. R-squared:                  0.329
Method:                 Least Squares   F-statistic:                     98.42
Date:                Sat, 19 Oct 2024   Prob (F-statistic):           4.35e-19
Time:                        11:04:29   Log-Likelihood:                -573.34
No. Observations:                 200   AIC:                             1151.
Df Residuals:                     198   BIC:                             1157.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          9.3116      0.563     16.542      0.000       8.202      10.422
Radio          0.2025      0.020      9.921      0.000       0.162       0.243
==============================================================================
Omnibus:                       19.358   Durbin-Watson:                   1.946
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               21.910
Skew:                          -0.764   Prob(JB):                     1.75e-05
Kurtosis:                       3.544   Cond. No.                         51.4
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Result Interpretation
The coefficient value for Radio is  0.2025, which means that for every additional unit spent on radio advertising (e.g., one dollar or another unit of measure), sales are expected to increase by 202.5 units (as values are in thousands)
___

## Regression between sales and Newspaper

In [12]:
# Defining the independent variable (Radio) and the dependent variable (Sales)
X = advertising['Newspaper']
y = advertising['Sales']

# Adding a constant to the independent variable (for the intercept)
X = sm.add_constant(X)

# Creating the linear regression model
model = sm.OLS(y, X).fit()

# Getting the model summary
model_summary = model.summary()
model_summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.052
Model:                            OLS   Adj. R-squared:                  0.047
Method:                 Least Squares   F-statistic:                     10.89
Date:                Sat, 19 Oct 2024   Prob (F-statistic):            0.00115
Time:                        11:05:10   Log-Likelihood:                -608.34
No. Observations:                 200   AIC:                             1221.
Df Residuals:                     198   BIC:                             1227.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         12.3514      0.621     19.876      0.000      11.126      13.577
Newspaper      0.0547      0.017      3.300      0.001       0.022       0.087
==============================================================================
Omnibus:                        6.231   Durbin-Watson:                   1.983
Prob(Omnibus):                  0.044   Jarque-Bera (JB):                5.483
Skew:                           0.330   Prob(JB):                       0.0645
Kurtosis:                       2.527   Cond. No.                         64.7
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Result Interpretation
The coefficient value for Newspaper is  0.0547, which means that for every additional unit spent on radio advertising (e.g., one dollar or another unit of measure), sales are expected to increase by 54.7 units (as values are in thousands)
___

## Regression between Sales, TV, Radio and Newspaper

In [13]:
# Defining the independent variables (TV, Radio, Newspaper) and the dependent variable (Sales)
X = advertising[['TV', 'Radio', 'Newspaper']]
y = advertising['Sales']

# Adding a constant to the independent variables (for the intercept)
X = sm.add_constant(X)

# Creating the linear regression model
model_multiple = sm.OLS(y, X).fit()

# Getting the model summary
model_multiple_summary = model_multiple.summary()
model_multiple_summary


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.896
Method:                 Least Squares   F-statistic:                     570.3
Date:                Sat, 19 Oct 2024   Prob (F-statistic):           1.58e-96
Time:                        11:06:10   Log-Likelihood:                -386.18
No. Observations:                 200   AIC:                             780.4
Df Residuals:                     196   BIC:                             793.6
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.9389      0.312      9.422      0.000       2.324       3.554
TV             0.0458      0.001     32.809      0.000       0.043       0.049
Radio          0.1885      0.009     21.893      0.000       0.172       0.206
Newspaper     -0.0010      0.006     -0.177      0.860      -0.013       0.011
==============================================================================
Omnibus:                       60.414   Durbin-Watson:                   2.084
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              151.241
Skew:                          -1.327   Prob(JB):                     1.44e-33
Kurtosis:                       6.332   Cond. No.                         454.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Result Interpretation
- The coefficient value for Radio is 0.1885, which means that for every additional unit spent on radio advertising (e.g., one dollar or another unit of measure), sales are expected to `increase by 188.5 units`, holding all other factors constant (in this case, TV and Newspaper advertising).
- Since the coefficient is positive, it `indicates that there is a direct relationship` between radio advertising and sales. As radio advertising spending increases, sales increase as well.
___

Here's how to interpret the **standard error**, **t-statistic**, and **p-value** for the **Radio** variable in the regression model:

### 1. **Standard Error (std err)**:
   - **Interpretation**: The standard error measures the variability or uncertainty of the coefficient estimate. For Radio, the standard error is **0.009**. This indicates how much the estimate of the coefficient (0.1885) is expected to vary due to random sampling variability.
   - **Lower standard error**: Implies more precise estimates of the coefficient.
   - **Higher standard error**: Suggests more uncertainty in the coefficient estimate.
   - **Context**: In this case, the low standard error indicates that the estimate of Radio's effect on sales is relatively precise.

### 2. **t-statistic (t)**:
   - **Formula**: \( t = \frac{\text{Coefficient}}{\text{Standard Error}} \)
   - **Interpretation**: The t-statistic tests whether the coefficient is significantly different from 0 (no effect). For Radio, the t-statistic is **21.893**. 
   - **Higher t-statistic**: Suggests stronger evidence against the null hypothesis (i.e., a larger difference from zero).
   - **Rule of thumb**: A t-statistic greater than about 2 in absolute value usually indicates a statistically significant effect.

### 3. **p-value**:
   - **Interpretation**: The p-value measures the probability of observing the effect (or something more extreme) if the null hypothesis (that the true coefficient is 0) were true. 
   - For Radio, the **p-value is < 0.001**, meaning the probability of observing such a strong relationship by chance is extremely small.
   - **Statistical significance**: A p-value less than 0.05 is often considered statistically significant, meaning that the effect of Radio on Sales is very unlikely to be due to random chance.

### Summary for Radio:
- **Std. Error (0.009)**: The estimate of the radio coefficient (0.1885) is very precise.
- **t-statistic (21.893)**: The effect of radio advertising on sales is very strong and statistically significant.
- **p-value (< 0.001)**: The relationship between radio advertising and sales is highly significant, providing strong evidence against the null hypothesis.